Framework/Beamline/test/DetectorInfoTest.h

```diff
@@ -253,6 +254,241 @@ class DetectorInfoTest : public CxxTest::TestSuite {
     info.setRotation(0, rot);
     TS_ASSERT_EQUALS(info.rotation(0).coeffs(), rot.normalized().coeffs());
   }
+
+  void test_scanCount() {
+    DetectorInfo info(PosVec(1), RotVec(1));
+    TS_ASSERT_EQUALS(info.scanCount(0), 1);
+  }
+
+  void test_scanInterval() {
+    DetectorInfo info(PosVec(1), RotVec(1));
+    TS_ASSERT_EQUALS(info.scanInterval({0, 0}),
+                     (std::pair<int64_t, int64_t>(0, 0)));
+  }
+
+  void test_setScanInterval() {
+    DetectorInfo info(PosVec(1), RotVec(1));
+    info.setScanInterval({0, 0}, {1, 2});
+    TS_ASSERT_EQUALS(info.scanInterval({0, 0}),
+                     (std::pair<int64_t, int64_t>(1, 2)));
+  }
+
+  void test_setScanInterval_failures() {
+    DetectorInfo info(PosVec(1), RotVec(1));
+    TS_ASSERT_THROWS(info.setScanInterval({0, 0}, {1, 1}), std::runtime_error);
+    TS_ASSERT_THROWS(info.setScanInterval({0, 0}, {2, 1}), std::runtime_error);
```

Comment:
Yes, you can use either `TS_ASSERT_THROWS_ASSERT` or `TS_ASSERT_THROWS_EQUAL`. It is not the nicest but it works.

For example:
``` cpp
TS_ASSERT_THROWS_EQUALS(info.setScanInterval(0, {1, 1}), const std::runtime_error &e, std::string(e.what()), "DetectorInfo: cannot set scan interval with start > end");
```